src/payment/onchain.rs

```diff
@@ -63,7 +72,32 @@ impl OnchainPayment {
 			crate::total_anchor_channels_reserve_sats(&self.channel_manager, &self.config);
 		let send_amount =
 			OnchainSendAmount::ExactRetainingReserve { amount_sats, cur_anchor_reserve_sats };
-		self.wallet.send_to_address(address, send_amount)
+		self.wallet.send_to_address(address, send_amount, fee_rate)
+	}
+
+	/// Send an on-chain payment to the given address.
+	///
+	/// This will respect any on-chain reserve we need to keep, i.e., won't allow to cut into
+	/// [`BalanceDetails::total_anchor_channels_reserve_sats`].
+	///
+	/// If `fee_rate` is set it will used on the resulting transaction. Otherwise a reasonable
+	/// we'll retrieve an estimate from the configured chain source.
+	///
+	/// [`BalanceDetails::total_anchor_channels_reserve_sats`]: crate::BalanceDetails::total_anchor_channels_reserve_sats
+	#[cfg(feature = "uniffi")]
+	pub fn send_to_address(
+		&self, address: &bitcoin::Address, amount_sats: u64, fee_rate: Option<Arc<FeeRate>>,
+	) -> Result<Txid, Error> {
+		let rt_lock = self.runtime.read().unwrap();
+		if rt_lock.is_none() {
+			return Err(Error::NotRunning);
+		}
+
+		let cur_anchor_reserve_sats =
+			crate::total_anchor_channels_reserve_sats(&self.channel_manager, &self.config);
+		let send_amount =
+			OnchainSendAmount::ExactRetainingReserve { amount_sats, cur_anchor_reserve_sats };
+		self.wallet.send_to_address(address, send_amount, fee_rate.map(|f| f.0))
```

Comment:
Would it be worth DRYing up these method definitions with macro using a parameter for accessing the fee rate?